### ISSUE TITLE
Support for fetching historical proofs for finalised blocks

### ISSUE THREAD

**Issue by TheMj0ln1r** [ChainSafe/lodestar#7316]
### Problem description

Lodestar prover endpoints are generating historical proofs for optimistic blocks but not for the Finalised blocks. 

### Solution description

Should be able to generate historical proofs for the older blocks(finalised).

**Comment by nazarhussain** [comment]
@TheMj0ln1r Would you clarify which element you are referring to. The `prover` which is our lightclient proxy client for web3 providers. Orr the `proof api` a.k.a `/eth/v0/beacon/proof/block/{block_id}`?

**Comment by TheMj0ln1r** [comment]
Yes @nazarhussain , issue is with `getStateProof` in the `ApiClient` of `@lodestar/api`. There if we pass a state_root of an optimistic block the proof is generating but not if we pass a finalised block. I've tested that `getStateProof` is generating proofs for last 62 blocks not more than that.

**Comment by nflaig** [comment]
See [discord](https://discord.com/channels/593655374469660673/1197575814494035968/1366665850555207720) for more discussion on this, created a [thread](https://discord.com/channels/593655374469660673/1366804621242404968) for any future discussion
